spark3/src/test/java/org/apache/iceberg/actions/TestCreateActions.java

```diff
@@ -0,0 +1,272 @@
+            "type", "hive",
+            "default-namespace", "default",
+            "parquet-enabled", "true",
+            "cache-enabled", "false" // Spark will delete tables using v1, leaving the cache out of sync
+        )},
+        new Object[] { "testhive", SparkCatalog.class.getName(), ImmutableMap.of(
+            "type", "hive",
+            "default-namespace", "default"
+        )}
+    };
+  }
+
+  @Rule
+  public TemporaryFolder temp = new TemporaryFolder();
+
+  String baseTableName = "baseTable";
+  File tableDir;
+  String tableLocation;
+  final String implementation;
+  final String type;
+  final TableCatalog catalog;
+
+  public TestCreateActions(
+      String catalogName,
+      String implementation,
+      Map<String, String> config) {
+    super(catalogName, implementation, config);
+    this.catalog = (TableCatalog) spark.sessionState().catalogManager().catalog(catalogName);
+    this.implementation = implementation;
+    this.type = config.get("type");
+  }
+
+  @Before
+  public void before() {
+    try {
+      this.tableDir = temp.newFolder();
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+    this.tableLocation = tableDir.toURI().toString();
+
+    spark.conf().set("hive.exec.dynamic.partition", "true");
+    spark.conf().set("hive.exec.dynamic.partition.mode", "nonstrict");
+    spark.sql(String.format("DROP TABLE IF EXISTS %s", baseTableName));
+
+
+    List<SimpleRecord> expected = Lists.newArrayList(
+        new SimpleRecord(1, "a"),
+        new SimpleRecord(2, "b"),
+        new SimpleRecord(3, "c")
+    );
+
+    Dataset<Row> df = spark.createDataFrame(expected, SimpleRecord.class);
+
+    df.select("id", "data").orderBy("data").write()
+        .mode("append")
+        .option("path", tableLocation)
+        .saveAsTable(baseTableName);
+  }
+
+  @After
+  public void after() throws IOException {
+    // Drop the hive table.
+    spark.sql(String.format("DROP TABLE IF EXISTS %s", baseTableName));
+  }
+
+
+  @Test
+  public void testMigratePartitioned() throws Exception {
+    String dest = uniqueName("iceberg_migrate_partitioned");
+    String source = uniqueName("test_migrate_partitioned_table");
+    testCreate(source,
+        dest,
+        "CREATE TABLE %s using parquet PARTITIONED BY (id) location '%s' AS SELECT * FROM %s",
+        () -> Actions.migrate(source).as(dest),
+        3);
+  }
+
+  @Test
+  public void testMigrateUnpartitioned() throws Exception {
+    String dest = uniqueName("iceberg_migrate_unpartitioned");
+    String source = uniqueName("test_migrate_unpartitioned_table");
+    testCreate(source,
+        dest,
+        "CREATE TABLE %s using parquet location '%s' AS SELECT * FROM %s",
+        () -> Actions.migrate(source).as(dest),
+        2);
+  }
+
+  @Test
+  public void testMigrateReplace() throws Exception {
+    // We can't do a replacement unless we have overridden the spark_catalog
+    if (catalog.name().equals("spark_catalog")) {
```

Comment:
Can you use `Assume` for this instead of `if`? That way it shows up as a skipped case.

Reply:
Yep, just learned about Assume last week. Will implement it here!